python/pyspark/sql/functions.py

```diff
@@ -123,7 +123,7 @@ def _create_judf(self):
         pickled_command, broadcast_vars, env, includes = _prepare_for_python_RDD(sc, command, self)
         ssql_ctx = sc._jvm.SQLContext(sc._jsc.sc())
         jdt = ssql_ctx.parseDataType(self.returnType.json())
-        judf = sc._jvm.UserDefinedPythonFunction(f.__name__, bytearray(pickled_command), env,
+        judf = sc._jvm.UserDefinedPythonFunction(f.__repr__(), bytearray(pickled_command), env,
```

Comment:
Could we try to get the name from it? Having a friendly name will help debugging.

name = f.__name__ if hasattr(f, “__name__") else f.__class__.__name__